### PR TITLE
Add more integration tests

### LINKED ISSUE #14: Write tests for node creators with interesting manifests

### LINKED ISSUE #31: Write integration tests
- [x] boolean-string
- [x] bower-npm-names-no-match
- [x] bower-origami-description-no-match
- [x] brand-not-string
- [x] brand-not-valid
- [x] brands-not-array
- [x] browser-feature-not-string
- [x] browser-feature-set-not-array
- [x] browser-features-not-object
- [x] browser-features-wrong-shape
- [x] category-not-valid
- [x] ci
- [x] demo-brands-not-array
- [x] demo-classes-not-string
- [x] demo-data-bad-file
- [x] demo-data-not-file-or-object
- [x] demo-deps-not-array
- [x] demo-deps-not-strings
- [x] demo-js-not-file
- [x] demo-sass-not-file
- [x] demo-template-not-file
- [x] demos-not-array
- [x] description-not-string
- [x] keyword-not-string
- [x] keywords-not-array
- [x] name-not-ascii
- [x] name-not-string
- [x] non-file-main
- [x] non-o-prefix
- [x] not-a-boolean
- [x] not-string
- [x] npm-origami-description-no-match
- [x] origami-type-is-module
- [x] origami-type-not-valid
- [x] origami-version-not-a-number
- [x] status-not-valid
- [x] support-email-not-email
- [x] support-email-not-ft
- [x] support-email-not-string
- [x] support-not-string
- [x] support-not-url
- [x] support-slack-not-org-channel
- [x] support-slack-not-string
- [x] referenced-missing-main.js
- [x] referenced-missing-main.scss
- [x] unreferenced-existing-main.scss
- [x] unreferenced-existing-main.js

i'm thinking of a structure like:

```tree
integration-tests/
├── default-template
│   ├── component
│   │   ├── bower.json
│   │   ├── main.js
│   │   ├── main.scss
│   │   ├── origami.json
│   │   └── package.json
│   └── expected.json
└── wrong-bower-name
    ├── component
    │   ├── bower.json
    │   └── origami.json
    └── expected.json
```

where `expected.json` is a file containing something like:

```json
{
  "exit": 0
}
```

```json
{
  "exit": 1,
  "warnings": [],
  "errors": ["wrong-bower-name"]
}
```

and a little node-tap script that iterates through all those and makes sure running the program on them returns the correct results. something like that.

### PR DESCRIPTION
**Pull request by chee**
closes #31, and in fact also resolves #14

### CODE REVIEW

**Comment by chee** [comment]
sorry about the heft of this hungus

File: integration-tests/bad-browser-feature-sets/component/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# default-template
+
+This is the default component generated by the
```
Comment (notlee) [inline integration-tests/bad-browser-feature-sets/component/README.md]: I think this readme lies

File: integration-tests/browser-features-not-object/component/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# default-template
+
+This is the default component generated by the
```
Comment (notlee) [inline integration-tests/browser-features-not-object/component/README.md]: I think this readme lies

File: integration-tests/very-poor-demos/component/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# default-template
+
+This is the default component generated by the
```
Comment (notlee) [inline integration-tests/very-poor-demos/component/README.md]: I think this readme lies

File: integration-tests/category-not-valid/component/origami.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+	"description": "description.",
+	"keywords": ["keyword"],
+	"origamiType": "component",
+	"origamiCategory": "yeet",
```
Comment (notlee) [inline integration-tests/category-not-valid/component/origami.json:5]: new category poposal? 😁

File: integration-tests/ci-in-error/component/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# ci
+
+this fails because the CI field is deprecated
```
Comment (notlee) [inline integration-tests/ci-in-error/component/README.md:3]: useful 👍 I wasn't sure at first

File: integration-tests/integration-test-base/component/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# perfect with js, sass and demos
+
```
Comment (notlee) [inline integration-tests/integration-test-base/component/README.md]: should this replace `integration-tests/default-template`?

File: integration-tests/js-present-but-unreferenced/component/README.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# perfect with js, sass and demos
+
+This one should pass
```
Comment (notlee) [inline integration-tests/js-present-but-unreferenced/component/README.md]: this readme lies

File: integration-tests/js-referenced-but-missing/component/README.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# perfect with js, sass and demos
+
+This one should pass
```
Comment (notlee) [inline integration-tests/js-referenced-but-missing/component/README.md]: this readme lies

File: integration-tests/sass-referenced-but-missing/component/README.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# perfect with js, sass and demos
+
```
Comment (notlee) [inline integration-tests/sass-referenced-but-missing/component/README.md:2]: this readme lies

File: integration-tests/status-not-valid/component/origami.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+	"origamiType": "component",
+	"origamiCategory": "components",
+	"origamiVersion": 1,
+	"brands": ["master", "internal", "whitelabel"],
+	"support": "https://github.com/Financial-Times/default-template/issues",
+	"supportContact": {
+		"email": "chee.rabbits@ft.com",
+		"slack": "financialtimes/#origami-chat"
+	},
+	"supportStatus": "incubating"
```
Comment (notlee) [inline integration-tests/status-not-valid/component/origami.json:13]: 😂 😱

File: integration-tests/support-email-not-email/component/origami.json [hunk 11]
```diff
@@ -0,0 +1,14 @@
+{
+	"description": "description.",
+	"keywords": ["keyword"],
+	"origamiType": "component",
+	"origamiCategory": "components",
+	"origamiVersion": 1,
+	"brands": ["master", "internal", "whitelabel"],
+	"support": "https://github.com/Financial-Times/default-template/issues",
+	"supportContact": {
+		"email": "chee.rabbits AT ft.com",
```
Comment (notlee) [inline integration-tests/support-email-not-email/component/origami.json:10]: This test has no github errors, should it or are we testing that we continue to allow anything? The readme suggests not but I wasn't sure:
>we define **valid email** as `anything@anything`.

**Review by notlee — COMMENTED** [review]
Nice work! :D This was a big one to review. Most of my comments are just pointing to readmes that need updating, I think there was only one test which looked like it might not be testing the right thing

File: integration-tests/integration-test-base/component/README.md (hunk 6 above)
Comment (chee) [reply integration-tests/integration-test-base/component/README.md]: ah, i was going to use this one as the source for a script for generating new tests but never did

File: integration-tests/support-email-not-email/component/origami.json (hunk 11 above)
Comment (chee) [reply integration-tests/support-email-not-email/component/origami.json:10]: oh yes, this one rightly fails and the `github` file is wrong!

**Comment by chee** [comment]
thank you for the review, @notlee !

**Review by JakeChampion — APPROVED** [review]

**Review by notlee — APPROVED** [review]

**Pull request merged**